test/gtest/ucs/test_sock.cc

```diff
@@ -674,6 +719,48 @@ UCS_TEST_F(test_socket, sockaddr_cmp_err) {
                           (const struct sockaddr*)&sa_in);
 }
 
+UCS_TEST_F(test_socket, subnet_match) {
```

Comment:
maybe add similar tests for case where af matches but subnet itself does not match?